src/mathematics/glyphinfo.rs

```diff
@@ -0,0 +1,142 @@
+    pub GlyphInfo { // MathGlyphInfo
+        italics_correction_offset      (u16), // MathItalicsCorrectionInfo
+        top_accent_offset              (u16), // MathTopAccentAttachment
+        extended_shape_coverage_offset (u16), // ExtendedShapeCoverage
+        kern_info_offset               (u16), // MathKernInfo
+
+        italics_corrections (Option<ItalicsCorrections>) |this, tape, position| {
+            jump_take_maybe!(tape, position, this.italics_correction_offset)        
+        },
+
+        top_accent_attachments (Option<TopAccentAttachments>) |this, tape, position| {
+            jump_take_maybe!(tape, position, this.top_accent_offset)
+        },
+
+        extended_shape_coverage (Option<Coverage>) |this, tape, position| {
+            jump_take_maybe!(tape, position, this.extended_shape_coverage_offset)
+        },
+
+        kern_info (Option<KernInfo>) |this, tape, position| {
+            jump_take_maybe!(tape, position, this.kern_info_offset)
+        },
+    }
+}
+
+table! {
+    @position
+    #[doc = "A table of italics corrections."]
+    pub ItalicsCorrections { // MathItalicsCorrectionInfo
+        coverage_offset (u16), // Coverage
+        count           (u16), // ItalicsCorrectionCount
+
+        corrections (Vec<ValueRecord>) |this, tape, _| { // ItalicsCorrection
+           tape.take_given(this.count as usize)
+        },
+
+        coverage (Coverage) |this, tape, position| {
+            jump_take!(tape, position, this.coverage_offset)
+        },
+    }
+}
+
+table! {
+    @position
+    #[doc = "A table of horizontal positioning for top accents."]
+    pub TopAccentAttachments { // MathTopAccentAttachment
+        coverage_offset (u16), // TopAccentCoverage
+        count           (u16), // TopAccentAttachmentCount
+
+        attachments (Vec<ValueRecord>) |this, tape, _| {    
+           tape.take_given(this.count as usize)
+        },
+
+        coverage (Coverage) |this, tape, position| {
+            jump_take!(tape, position, this.coverage_offset)
+        },
+    }
+}
+
+table! {
+    @position
+    #[doc = "A table of kerning information for mathematical glyphs."]
+    pub KernInfo {
+        coverage_offset (u16), // MathKernCoverage
+        count           (u16), // MathKernCount
+
+        kernings (Vec<KernGlyph>) |this, tape, position| {
+            let mut kern_glyphs: Vec<KernGlyph> = Vec::with_capacity(this.count as usize);
+            for _ in 0..this.count {
+                kern_glyphs.push(try!(tape.take()));
+            }
+            for kern in &mut kern_glyphs {
+                kern.top_right = jump_take_maybe!(@unwrap tape, position, kern.top_right_offset);
+                kern.top_left = jump_take_maybe!(@unwrap tape, position, kern.top_left_offset);
+                kern.bottom_right = jump_take_maybe!(@unwrap tape, position, kern.bottom_right_offset);
+                kern.bottom_left = jump_take_maybe!(@unwrap tape, position, kern.bottom_left_offset);
+            }
```

Comment:
It’s what the `Walue` trait is for. I think it should be possible to resolve this case by implementing `Walue<u64>` for `KernGlyph` (`Kerning`) where the parameter is the needed position. So you can try to change

```rust
kern_glyphs.push(try!(tape.take()))
```

to something like

```rust
values.push(try!(tape.take_given(position)))
```

There are many such cases in the code. Have a look at [Mark1](https://github.com/bodoni/opentype/blob/master/src/glyph_positioning/element.rs#L474), for instance.